lib/test_formulae.rb

```diff
@@ -97,33 +97,48 @@ def download_artifact_from_previous_run!(name)
         response = GitHub::API.open_rest(url)
         return if response.fetch("total_count").zero?
 
-        wanted_artifact = response.fetch("artifacts").find { |artifact| artifact.fetch("name") == name }
+        artifacts = response.fetch("artifacts")
+        wanted_artifact = artifacts.find { |artifact| artifact.fetch("name") == name }
+        wanted_artifact ||= artifacts.find { |artifact| artifact.fetch("name") == "event_payload" }
         return if wanted_artifact.blank?
 
-        ohai "Downloading #{name} from #{previous_github_sha}"
+        wanted_artifact_name = wanted_artifact.fetch("name")
+        cached_event_json&.unlink if wanted_artifact_name == "event_payload"
+
+        ohai "Downloading #{wanted_artifact_name} from #{sha}"
         download_url = wanted_artifact.fetch("archive_download_url")
 
         artifact_cache.mkpath
         artifact_cache.cd do
           GitHub.download_artifact(download_url, run_id)
         end
+
+        return if wanted_artifact_name == name
+
+        download_artifact_from_previous_run!(name)
```

Comment:
This doesn't seem safe. We could get an event payload artifact from a different run from the bottles.

I feel like it's just going to be safer to just bundle the event payload (or a subset of it) in with the bottle artifact.